### ISSUE TITLE
Prisma Client: updating to 4.16.0 or 4.16.1 breaks Cloudflare worker, it errors with `The package "path" wasn't found on the file system but is built into node`

### ISSUE THREAD

**Issue by osejrafael** [prisma/prisma#19972]
### Bug description

When you update prisma to the latest version (4.16.1) it breaks `wrangler`.

The error is:
```bash
  node_modules/@prisma/client/scripts/default-index.js:1:21:
      1 │ const path = require('path')
  
  The package "path" wasn't found on the file system but is built into node
```

A workaround is to set the `node_compat = true` flag in the wrangler.toml file.
However this comes with some downfalls.

### How to reproduce

1. Install prima 4.16.1
2. Generate edge client with the --data-proxy flag
3. Setup prisma using the edge client

### Expected behavior

Edge client works in edge environments. 

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
  previewFeatures = ["jsonProtocol"]
}

datasource db {
  provider = "mysql"
  url = env("DATABASE_URL")
  relationMode = "prisma"
}
```

```ts
import { PrismaClient } from "@prisma/client/edge";

export default function prisma(url: string) {
  const client = new PrismaClient({
    datasources: {
      db: {
        url,
      },
    },
  });

  return client;
}

export type ExtendedPrismaClient = ReturnType<typeof prisma>;

```

### Environment & setup

- OS: macOS
- Database: PlanetScale
- Node.js version: 18.16

### Prisma Version

```
4.16.1
```

**Comment by Jolg42** [comment]
Note: 
Looking at `@prisma/client/scripts/default-index.js` https://github.com/prisma/prisma/blob/main/packages/client/scripts/default-index.js#L1

We recently (since `4.16.0`) added this line in https://github.com/prisma/prisma/pull/19696 which matches the error here
```
const path = require('path')
```

**Comment by Jolg42** [comment]
@osejrafael Thanks for the report!

Are you using a Prisma Client extension? (Accelerate maybe?)

**Comment by osejrafael** [comment]
@Jolg42 hey!
I was in fact using accelerate... However I removed it and regenerated the proxy client, still same issue. 
If you think the problem is with accelerate, I can try again later tonight by rebuilding the package-lock and reinstalling dependencies

**Comment by Jolg42** [comment]
I think I have enough to try a reproduction, I already identified the problem, but I want to make sure we can reproduce it and also test it.

**Comment by Jolg42** [comment]
Note: tests are in https://github.com/prisma/ecosystem-tests/pull/3669

And I verified they do catch this error in a manual run using 4.16.2 https://github.com/prisma/ecosystem-tests/pull/3674
```
 ⛅️ wrangler 3.1.1 (update available 3.1.2)
-------------------------------------------
✘ [ERROR] Could not resolve "path"
    node_modules/.pnpm/@prisma+client@4.16.2_prisma@4.16.2/node_modules/@prisma/client/scripts/default-index.js:1:21:
      1 │ const path = require('path')
        ╵                      ~~~~~~
  The package "path" wasn't found on the file system but is built into node.
  Add "node_compat = true" to your wrangler.toml file to enable Node.js compatibility.
```
